devito/function.py

```diff
@@ -477,8 +479,8 @@ def _arg_defaults(self, alias=None):
         args = ReducerMap({key.name: self._data_buffer})
 
         # Collect default dimension arguments from all indices
-        for i, s, o, k in zip(self.indices, self.shape, self.staggered, key.indices):
-            args.update(i._arg_defaults(start=0, size=s+o, alias=k))
+        for i, s, o in zip(key.indices, self.shape, self.staggered):
+            args.update(i._arg_defaults(start=0, size=s+o))
```

Comment:
I found out it was useless, so I removed it. This is because in the current implementation we work out aliases in `_arg_values`, and in turn `_arg_values` calls `_arg_defaults` either on `self` or `alias`. Explanation a bit contrived, but in practice it's much simpler

Reply:
Ok